internal/middleware/user.go

```diff
@@ -1,24 +1,81 @@
 package middleware
 
 import (
+	"fmt"
+	"net/http"
+	"time"
+
 	"github.com/gin-gonic/gin"
+	"github.com/hashicorp/golang-lru/v2/expirable"
+	"github.com/rs/zerolog/log"
+	"github.com/ucl-arc-tre/portal/internal/graceful"
+	"github.com/ucl-arc-tre/portal/internal/rbac"
 	"github.com/ucl-arc-tre/portal/internal/types"
+	"gorm.io/gorm"
 )
 
 const (
+	userCacheTTL      = 1 * time.Minute
```

Comment:
thoughts appreciated on this number. _probably_ safe to be longer

Reply:
does this mean it refreshes the cache every minute?

Reply:
it means that any object in the cache can be there for a maximum of a minute

Reply:
ah, yes certainly longer but unsure how long exactly